src/service.js

```diff
@@ -111,22 +111,31 @@ export function getExistingServiceInEmbedScope(win, id, opt_fallbackToTopWin) {
  * @return {Object} The service.
  */
 export function getExistingServiceForDocInEmbedScope(
-  nodeOrDoc, id, opt_fallbackToTopWin) {
-  // First, try to resolve via local (embed) window.
-  if (nodeOrDoc.nodeType) {
-    // If a node is passed, try to resolve via this node.
-    const win = toWin(/** @type {!Document} */ (
-      nodeOrDoc.ownerDocument || nodeOrDoc).defaultView);
-    const local = getLocalExistingServiceForEmbedWinOrNull(win, id);
-    if (local) {
-      return local;
-    }
+  nodeOrDoc, id, opt_fallbackToTopWin)
+{
+  const isAmpDoc = !nodeOrDoc.nodeType;
+  // If an ampdoc is passed, resolve via ampdoc.
+  if (isAmpDoc) {
+    const ampdoc =
+      /** @type {!./service/ampdoc-impl.AmpDoc} */ (nodeOrDoc);
+    return getServiceForDoc(ampdoc, id);
+  }
+  // Now nodeOrDoc must be a node.
+  // First, try to resolve via local embed window.
+  const document =
+    /** @type {!Document} */ (nodeOrDoc.ownerDocument || nodeOrDoc);
+  const win = toWin(document.defaultView);
+  const local = getLocalExistingServiceForEmbedWinOrNull(win, id);
+  if (local) {
+    return local;
   }
-  // If an ampdoc is passed or fallback is allowed, continue resolving.
-  if (!nodeOrDoc.nodeType || opt_fallbackToTopWin) {
-    return getServiceForDocDeprecated(nodeOrDoc, id);
+  // Next, node is embedded AND fallback is not allowed, return null.
+  // Otherwise, resolve via ampdoc.
+  const isEmbed = win != getTopWindow(win);
```

Comment:
If the section right above is about "local embed window", why not then just check `const isEmbed = ...; if (isEmbed...)` before calling the `getLocalExistingServiceForEmbedWinOrNull` above?